adafruit_ads1x15/ads1x15.py

```diff
@@ -317,3 +402,35 @@ def _read_register(self, reg: int, fast: bool = False) -> int:
             else:
                 i2c.write_then_readinto(bytearray([reg]), self.buf, in_end=2)
         return self.buf[0] << 8 | self.buf[1]
+
+    def read_config(self) -> None:
+        """Reads Config Register and sets all properties accordingly"""
```

Comment:
Why do you want this? The constructor sets everything to defaults that should override this.

Reply:
You could use this read_config function to get the configuration settings that are actually set in the ADC. Not necessarily to be used at startup, but rather if the chip has been on a while and perhaps the micro got out of sync with the ADC and you wanted to know what the ADC is doing.

Reply:
I think it'd be confusing to have the device state different from the state of the driver. I think it's better to assume you set it every time. That way it doesn't depend on the state of the device on start up.

Reply:
I certainly agree that it is confusing if the driver and device are in different states. However, currently the code only sends the configuration settings when doing a read, so it would only send settings at startup if you do a read at startup. I think this is a problem that would require even more changes to fix. My read_config function here just keeps you from getting into trouble by assuming the device is in a certain state when it's really not in that state. I'm not suggesting to rely on read_config at startup for configuration.

My use case for this function was I ran into a situation where the controller was reset, but the ADC was not. Since I didn't do a read at startup, the driver now had "default" settings, but the ADC had the settings from the last session. This led to unexpected results.

Reply:
> My use case for this function was I ran into a situation where the controller was reset, but the ADC was not. Since I didn't do a read at startup, the driver now had "default" settings, but the ADC had the settings from the last session. This led to unexpected results.

Ah! Should you write the default settings to the device on startup? That does sound confusing. I don't want to have this function though because it forces one to think about the difference between the device and the driver.